packages/app-mobile/root.tsx

```diff
@@ -646,6 +672,20 @@ class AppComponent extends React.Component {
 				state: 'initializing',
 			});
 
+			try {
+				this.unsubscribeNetInfoHandler_ = NetInfo.addEventListener(({ type, details }) => {
+					const isMobile = details.isConnectionExpensive || type === 'cellular';
+					reg.setIsOnMobileData(isMobile);
+					this.props.dispatch({
+						type: 'UPDATE_MOBILE_DATA_WARNING',
+						isOnMobileData: isMobile,
+					});
+				});
+			} catch (error) {
+				reg.logger().warn('Something went wrong while checking network info');
+				reg.logger().info(error);
+			}
```

Comment:
When the app starts, is this event handler going to be called immediately?

Reply:
Yes, that is guaranteed by the library.

> Your listener will be called with the latest information soon after you subscribe and then with any subsequent changes afterwards.

[docs](https://github.com/react-native-netinfo/react-native-netinfo#addeventlistener)

Reply:
Great, thanks for checking. Could you add a comment next to the code with that info please? We can always check the doc of course, but it's good to mention this slightly unusual behaviour directly in the code too.